### PR TITLE
Ci and packaging

### PR DESCRIPTION
**Pull request by Stefan8609**
Added the yaml file to implement continuous integration: this allows for the easy implementation of testing in the future.

Also set up a toml file that installs relevant dependencies - this file should be updated with new dependencies as our project progresses.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+[project]
+name = "example"
+version = "0.1.0"
+requires-python = ">=3.10"
+
+[project.dependencies]
+numpy = "^1.21.0"
+matplotlib = "^3.4.0"
+
```
Comment (napoleonpempena) [inline pyproject.toml]:
```suggestion
  [project]
  dependencies = [
      "numpy>=1.20"
      "matplotlib>=3.4.0"
  ]
```

**Review by napoleonpempena — CHANGES_REQUESTED** [review]
Suggested alternative syntax for project dependencies

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: CI
```
Comment (mr-gabrielrios) [inline .github/workflows/ci.yaml:1]: Adding version name for future version control documentation.
```suggestion
name: CI
version: 1.0
```

File: pyproject.toml (hunk 1 above)
Comment (mr-gabrielrios) [inline pyproject.toml]: Suggesting packages that will likely be used for data access from websites and establishing connection to HTTP pages.
```suggestion
"beautifulsoup4>=4.12.0"
"requests>=2.30.0"
```

Comment (GEODONG) [reply pyproject.toml]: I added several other libraries for data-processing

```suggestion
[project.dependencies]
numpy = "^1.21.0"
matplotlib = "^3.4.0"
pandas = "2.2.0"
json = "1.6.3"
 
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "${{ matrix.python-version }}"
+
+      - name: Install package and test deps
+        run: pip install -e .[test]
```
Comment (GEODONG) [inline .github/workflows/ci.yaml]: Change the installation of the pytest
```suggestion
        run: pip install pytest
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  tests:
+    name: "Tests on Python ${{ matrix.python-version }} for ${{ matrix.runs-on }}"
+    runs-on: "${{ matrix.runs-on }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.12"]
+        runs-on: ["ubuntu-latest", "macos-latest", "windows-latest"]
```
Comment (GEODONG) [inline .github/workflows/ci.yaml:18]: I think we can have a smaller size of matrix.
```suggestion
        runs-on: ["ubuntu-latest", "macos-latest"]
```

**Review by GEODONG — COMMENTED** [review]
Feat: change the size of the matrix and update the approach to install the pytest library.

**Pull request merged**